CS3230Project/View/Checkup.cs

```diff
@@ -352,25 +353,32 @@ private void SubmitDiagnosis_Click(object sender, EventArgs e)
 
         private void DiagnosisDataGridView_CellClick(object sender, DataGridViewCellEventArgs e)
         {
-            if (!this.finalDiagnosisExists)
+            try
             {
-                var rowIndex = e.RowIndex;
-                var diagnosisId = (int)this.diagnosisDataGridView.Rows[rowIndex].Cells[0].Value;
-                var diagnoses = DiagnosisManagerViewModel.GetDiagnoses(this.appointmentId);
-                Diagnosis diagnosis = null;
-
-                foreach (var currDiagnosis in diagnoses)
+                if (!this.finalDiagnosisExists)
                 {
-                    if (currDiagnosis.DiagnosisId == diagnosisId)
+                    var rowIndex = e.RowIndex;
+                    var diagnosisId = (int)this.diagnosisDataGridView.Rows[rowIndex].Cells[0].Value;
+                    var diagnoses = DiagnosisManagerViewModel.GetDiagnoses(this.appointmentId);
+                    Diagnosis diagnosis = null;
+
+                    foreach (var currDiagnosis in diagnoses)
                     {
-                        diagnosis = currDiagnosis;
+                        if (currDiagnosis.DiagnosisId == diagnosisId)
+                        {
+                            diagnosis = currDiagnosis;
+                        }
                     }
-                }
 
-                var modifyDiagnosisDialog = new ModifyDiagnosis(diagnosis, this.appointmentId);
-                modifyDiagnosisDialog.DiagnosisSubmittedEvent += this.DiagnosisSubmitEvent;
-                modifyDiagnosisDialog.ShowDialog();
+                    var modifyDiagnosisDialog = new ModifyDiagnosis(diagnosis, this.appointmentId);
+                    modifyDiagnosisDialog.DiagnosisSubmittedEvent += this.DiagnosisSubmitEvent;
+                    modifyDiagnosisDialog.ShowDialog();
+                }
             }
+            catch (Exception)
+            {
```

Comment:
Won't you need to alert the user on an exception if the submission fails. You want to have something in that catch block

Reply:
Fixed. I removed the Try/Catch block, since I only added to catch the exception for someone clicking the header to sort the results. I know just added a check in the if statement to prevent the exception.